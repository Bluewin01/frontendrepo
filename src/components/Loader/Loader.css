@keyframes spinny {
  0% {
    transform: rotate(0deg);
    background: #f58888;
  }
  50% {
    transform: rotate(180deg);
    background: #fe0808;
  }
  100% {
    transform: rotate(360deg);
    background: #f58888;
  }
}

.spinner {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  text-align: center;
}

.spinner:before {
  content: "";
  height: 100%;
  display: inline-block;
  vertical-align: middle;
}

.spinner .spinner-frame {
  display: inline-block;
  vertical-align: middle;
  width: 100px;
  height: 100px;
  border-radius: 50%;
  position: relative;
  overflow: hidden;
  border: 5px solid #fff;
  padding: 10px;
}

.spinner .spinner-frame .spinner-cover {
  background: #fff;
  width: 100%;
  height: 100%;
  border-radius: 50%;
  position: relative;
  z-index: 2;
}

.spinner .spinner-frame .spinner-bar {
  background: #fff;
  width: 50%;
  height: 50%;
  position: absolute;
  top: 0;
  left: 0;
  border-radius: 100% 0 0 0;
  animation: spinny 2s linear infinite;
  transform-origin: 100% 100%;
}
