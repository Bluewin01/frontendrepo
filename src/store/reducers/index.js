import { combineReducers } from "redux";
import { configureStore } from "@reduxjs/toolkit";
import lang from "../Lang/LangReducer";
import loader from "../Loader/LoaderReducer";
import Auth from "../Auth/AuthReducer";
import Template from "../Template/TemplateReducer";

const rootReducer = combineReducers({
  lang,
  loader,
  Auth,
  Template,
});

export const setupStore = (preloadedState) => {
  return configureStore({
    reducer: rootReducer,
    preloadedState,
  });
};

export default rootReducer;
