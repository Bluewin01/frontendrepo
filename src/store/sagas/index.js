import { AuthSaga } from "../Auth/AuthSaga";
import { TemplateSaga } from "../Template/TemplateSaga";
import { fork, all } from "redux-saga/effects";

export function* watchSagas() {
  yield all([AuthSaga(), TemplateSaga()]);
}
