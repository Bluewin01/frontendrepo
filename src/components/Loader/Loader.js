import React from "react";
import "./Loader.css";

const Loader = () => {
  return (
    <div className="spinner">
      <div className="spinner-frame">
        <div className="spinner-cover"></div>
        <div className="spinner-bar"></div>
      </div>
    </div>
  );
};

export default Loader;
