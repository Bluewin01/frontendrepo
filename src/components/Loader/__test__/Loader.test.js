import React from "react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import store from "../../../store";
import { Provider } from "react-redux";
import Loader from "../Loader";

describe("Component: Loader", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });
  it("should be render correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Loader />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
