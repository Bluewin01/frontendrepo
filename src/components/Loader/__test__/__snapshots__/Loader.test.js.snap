// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Loader should be render correctly 1`] = `
<div
  className="spinner"
>
  <div
    className="spinner-frame"
  >
    <div
      className="spinner-cover"
    />
    <div
      className="spinner-bar"
    />
  </div>
</div>
`;
